megamek/src/megamek/server/Server.java

```diff
@@ -1881,6 +1881,37 @@ private void resetEntityPhase(IGame.Phase phase) {
         game.clearIlluminatedPositions();
         send(new Packet(Packet.COMMAND_CLEAR_ILLUM_HEXES));
     }
+    
+    /*
+     *  Called during the end phase. Checks each entity for ASEW effects counters and decrements them by 1 if > 0
+     */
+    
+    public void decrementASEWTurns() {
+        for (Iterator<Entity> e = game.getEntities(); e.hasNext(); ) {
+            final Entity entity = e.next();
+            // Decrement ASEW effects
+            if (entity instanceof Dropship) {
```

Comment:
The Dropship and Jumpship code seems identical. Why not coalesce it?

Also, you can check entity.getEntityType() & ETYPE_JUMPSHIP instead of using reflection to improve performance a bit.

Reply:
I separate dropship and jumpship Because the closest common ancestor that Dropship and Jumpship have in common is Aero, and Aero (or Small Craft) are not supposed to have the get/set ASEWAffected(loc) methods.    I thought that if I did something like "if ((entity instanceof Dropship) || (entity instanceof Jumpship)) {
Dropship d = (Dropship) entity;
Jumpship j = (Jumpship) entity;     that I would get a class cast exception

Reply:
I do not know how to use the entity.getEntityType() & ETYPE_JUMPSHIP  in Server- it gives me errors

Reply:
Also, if I did that, will the ETYPE still apply to the Warship and Space Station children of Jumpship?

Reply:
In that case, I think if you want to check for jumpship and all its child types, you probably want entity.getEntityType() & ETYPE_JUMPSHIP != 0, rather than entity.getEntityType() & ETYPE_JUMPSHIP == ETYPE_JUMPSHIP. I'm pretty sure ETYPE_WARSHIP is something like 0010001 (not the exact value, but something like it), while ETYPE_JUMPSHIP is just 0010000. Basically, with >= 0 you're seeing if a particular group of bits is set ignoring any other bits. With == ETYPE_WHATEVER, you're seeing if only that group of bits is set and nothing else.